packages/jsActions/web-nanoflow-actions/src/TakePicture.ts

```diff
@@ -96,15 +88,35 @@ export async function TakePicture(picture: mendix.lib.MxObject): Promise<boolean
 
             secondScreenCleanup();
 
-            resolve();
+            resolve(false);
         });
 
         switchControl.addEventListener("click", switchControlHandler);
 
-        actionControl.addEventListener("click", takePictureHandler);
+        actionControl.addEventListener(
+            "click",
+            showConfirmationScreen
+                ? takePictureHandler
+                : () => {
+                      const videoCanvas = getVideoCanvas();
+                      savePicture(videoCanvas, () => {
+                          videoCanvas.remove();
+                          closeControlHandler();
+                      });
+                  }
+        );
```

Comment:
Do we need to clean this up somewhere? I don't see it, but maybe i missed it

Nit: Could do one function and put the conditional in the function instead of the other way around.

Reply:
closeControlHandler(); remove the dom and the videoCanvas.remove releases the memory